cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/MaintenanceTimeWindowsTest.java

```diff
@@ -0,0 +1,308 @@
+                async.complete();
+            });
+
+        async.await();
+    }
+
+    @Test
+    public void testZkRollingUpdateNoMaintenance(TestContext context) {
+
+        Async async = context.async();
+
+        testZkRollingUpdate(null,
+            () -> Date.from(LocalDateTime.of(2018, 11, 26, 11, 00, 0).atZone(ZoneId.of("GMT")).toInstant()),
+            r -> {
+                String generation = getClusterCaGenerationPod(ZookeeperCluster.zookeeperPodName(NAME, 0));
+                context.assertEquals("1", generation, "Pod had unexpected generation " + generation);
+                async.complete();
+            });
+
+        async.await();
+    }
+
+    @Test
+    public void testZkRollingUpdateMoreMaintenanceWindowsSatisfied(TestContext context) {
+
+        Async async = context.async();
+
+        testZkRollingUpdate(Arrays.asList("* * 8-10 * * ?", "* * 14-15 * * ?"),
+            () -> Date.from(LocalDateTime.of(2018, 11, 26, 9, 00, 0).atZone(ZoneId.of("GMT")).toInstant()),
+            r -> {
+                String generation = getClusterCaGenerationPod(ZookeeperCluster.zookeeperPodName(NAME, 0));
+                context.assertEquals("1", generation, "Pod had unexpected generation " + generation);
+                async.complete();
+            });
+
+        async.await();
+    }
+
+    @Test
+    public void testKafkaRollingUpdateNoMaintenance(TestContext context) {
+
+        Async async = context.async();
+
+        testKafkaRollingUpdate(null,
+            () -> Date.from(LocalDateTime.of(2018, 11, 26, 11, 00, 0).atZone(ZoneId.of("GMT")).toInstant()),
+            r -> {
+                String generation = getClusterCaGenerationPod(KafkaCluster.kafkaPodName(NAME, 0));
+                context.assertEquals("1", generation, "Pod had unexpected generation " + generation);
+                async.complete();
+            });
+
+        async.await();
+    }
+
+    @Test
+    public void testKafkaRollingUpdateMaintenanceSatisfiedDifferentTz(TestContext context) {
+
+        Async async = context.async();
+
+        testZkRollingUpdate(Collections.singletonList("* * 14-15 * * ?"),
+            () -> Date.from(LocalDateTime.of(2018, 11, 26, 9, 00, 0).atZone(ZoneId.of("Pacific/Easter")).toInstant()),
+            r -> {
+                String generation = getClusterCaGenerationPod(ZookeeperCluster.zookeeperPodName(NAME, 0));
+                context.assertEquals("1", generation, "Pod had unexpected generation " + generation);
+                async.complete();
+            });
+
+        async.await();
+    }
+
+    private String getClusterCaGenerationPod(String podName) {
+        Pod pod = this.mockClient.pods().inNamespace(NAMESPACE).withName(podName).get();
+        return pod.getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION);
+    }
+
+    private void testZkRollingUpdate(List<String> maintenanceTimeWindows, Supplier<Date> dateSupplier,
```

Comment:
Might be handy to document what case this is testing, since it's not so clear from the method name.